app/assets/src/components/views/amr_heatmap/AMRHeatmapControls.jsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Divider } from "~/components/layout";
+import { Dropdown } from "~ui/controls/dropdowns";
+
+import cs from "./amr_heatmap_view.scss";
+
+export default class AMRHeatmapControls extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  generateChangeFunction(filter) {
+    const changeFunction = option => {
+      if (option === this.props.selectedOptions[filter]) {
+        return;
+      }
+      this.props.onSelectedOptionsChange({ [filter]: option });
+    };
+    return changeFunction;
+  }
+
+  renderFilterDropdowns() {
+    const filtersList = [];
+    this.props.filters.forEach((_, filter) => {
```

Comment:
it might be easier to just have `filters` as an object instead of a Map

Reply:
The difference is minimal and using a `Map` makes it easy to order the dropdowns, so I'll keep it a Map.

Reply:
I didn't know this about `Maps`! Very cool.